pandas/core/dtypes/cast.py

```diff
@@ -1333,3 +1334,13 @@ def maybe_cast_to_integer_array(arr, dtype, copy=False):
     if is_integer_dtype(dtype) and (is_float_dtype(arr) or
                                     is_object_dtype(arr)):
         raise ValueError("Trying to coerce float values to integers")
+
+
+def ensure_python_int(value: Union[int, Any]) -> int:
```

Comment:
I gave the wrong direction, can you put in core/dtypes/common (right below the other ensure_*)

and can you add a doc-string (I know its trivial)

the typing seems odd here, shouldn't this be Union[int, np.integer] ?

Reply:
I'm not quite sure. In principle a float or a np.floating and possibly others could for some subvalues pass the test (e.g. if value is 1.0). But ok, passing a float would be a code smell, so just as well make the programmer see passing those as typing errors.